Themed Submission Components

ZahraaChreim-Atmire opened the pull request:
## References


## Description
This PR makes the **SubmissionFormFooterComponent**, **SubmissionSectionContainerComponent**, and **SubmissionFormComponent** themeable.

## Instructions for Reviewers

List of changes in this PR:
* First, created `ThemedSubmissionFormFooterComponent`, `ThemedSubmissionSectionContainerComponent`, and `ThemedSubmissionFormComponent` and extended them from `ThemedComponent<OriginalComponent>`.
* Second, set the themed component selectors to use the format `ds-themed-${original-selector}`.
* Third, implemented `getComponentName`, `importThemedComponent`, and `importUnthemedComponent`.
* Fourth, added all `@Input()` and `@Output()` properties from the original components and overrode `inAndOutputNames` to properly register inputs and outputs.
* Fifth, replaced references to the original components with their themed counterparts in routing modules and templates.

**How to test or review the PR.**
* Create a themed version of any of the updated components (e.g., `SubmissionFormFooterComponent`) in a theme, such as the **client theme**.
* Customize the themed component to make it visually distinct (e.g., add a unique class or modify the template).
* Navigate to a page where the component is used and confirm that the themed version is being used instead of the default one.

## Checklist
_This checklist provides a reminder of what we are going to look for when reviewing your PR. You do not need to complete this checklist prior creating your PR (draft PRs are always welcome).
However, reviewers may request that you complete any actions in this list if you have not done so. If you are unsure about an item in the checklist, don't hesitate to ask. We're here to help!_

- [x] My PR is **created against the `main` branch** of code (unless it is a backport or is fixing an issue specific to an older branch).
- [x] My PR is **small in size** (e.g. less than 1,000 lines of code, not including comments & specs/tests), or I have provided reasons as to why that's not possible.
- [x] My PR **passes [ESLint](https://eslint.org/)** validation using `npm run lint`
- [x] My PR **doesn't introduce circular dependencies** (verified via `npm run check-circ-deps`)
- [ ] My PR **includes [TypeDoc](https://typedoc.org/) comments** for _all new (or modified) public methods and classes_. It also includes TypeDoc for large or complex private methods.
- [x] My PR **passes all specs/tests and includes new/updated specs or tests** based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [x] My PR **aligns with [Accessibility guidelines](https://wiki.lyrasis.org/display/DSDOC8x/Accessibility)** if it makes changes to the user interface.
- [ ] My PR **uses i18n (internationalization) keys** instead of hardcoded English text, to allow for translations.
- [x] My PR **includes details on how to test it**. I've provided clear instructions to reviewers on how to successfully test this fix or feature.
- [ ] If my PR includes new libraries/dependencies (in `package.json`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.
- [ ] If my PR includes new features or configurations, I've provided basic technical documentation in the PR itself.
- [ ] If my PR fixes an issue ticket, I've [linked them together](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).

FrancescoMolinaro commented:
Hi @ZahraaChreim-Atmire , thanks for the changes! I wanted to test it locally but i noticed that the build in prod mode is failing and the tests too.
Could you please have a look into it? Thanks in advance.

ZahraaChreim-Atmire commented:
> Hi @ZahraaChreim-Atmire , thanks for the changes! I wanted to test it locally but i noticed that the build in prod mode is failing and the tests too. Could you please have a look into it? Thanks in advance.

Hi @FrancescoMolinaro , I've fixed the issues with the build and tests. Could you please take another look? Thanks!

FrancescoMolinaro reviewed (changes requested):
Hi @ZahraaChreim-Atmire , many thanks for the changes.
The PR looks good now, I have added just a small change request because I think one of the selector of the new themed compoents is wrong.
Could you please have a look?

Thanks in advance.

ZahraaChreim-Atmire commented:
> Hi @ZahraaChreim-Atmire , many thanks for the changes. The PR looks good now, I have added just a small change request because I think one of the selector of the new themed compoents is wrong. Could you please have a look?
> 
> Thanks in advance.

Hi @FrancescoMolinaro, you're absolutely right, the selector was incorrect. I've updated it to `ds-themed-base-submission-section-container`. Let me know if anything else needs adjusting.

Thanks again!

tdonohue commented on src/app/submission/edit/submission-edit.component.spec.ts:
```diff
@@ -57,6 +57,8 @@ describe('SubmissionEditComponent Component', () => {
 
   beforeEach(waitForAsync(() => {
+    // Fix for missing CSS custom property
+    document.documentElement.style.setProperty('--bs-xl', '1200');
```
This looks very odd for a spec file.  Could you describe why this spec/test requires us to add a missing CSS property in order to work?

ZahraaChreim-Atmire replied:
> This looks very odd for a spec file. Could you describe why this spec/test requires us to add a missing CSS property in order to work?

@tdonohue Initially, the test was failing with `TypeError: Cannot read properties of undefined (reading '--bs-xl')`, which led me to add both the `--bs-xl` property and `fixture.destroy()` in the test setup as adding only the `--bs-xl` property still results in the same error. However, after rechecking, I found that just adding `fixture.destroy()` in `afterEach` is sufficient to fix the issue, so the CSS property addition is actually unnecessary.
I’ve now removed that part from the PR.

tdonohue reviewed (commented):
@ZahraaChreim-Atmire : Thanks for this update.  I tested this today & it all looks good.  But, before we merge it, I have a question about one line of code below.

tdonohue reviewed (approved):
👍 Thanks @ZahraaChreim-Atmire !  This looks good to me now.
